test/unit/trace-test.js

```diff
@@ -0,0 +1,239 @@
+    tracer.startSpan('Name', { childOf: rootSpan });
+    t.true(rootSpan.context().getTrace().findNode(rootSpan.getId())._nodes.length, 2);
+    t.end();
+});
+
+test('test recording', (t) => {
+    const recorder = new ListRecorder();
+    const tracer = new APMTracer({ recorder: recorder });
+    const rootSpan = tracer.startSpan('Name');
+    const child = tracer.startSpan('Name', { childOf: rootSpan });
+
+    child.finish();
+    t.deepEqual(recorder.getTraces().length, 0);
+
+    rootSpan.finish();
+    t.deepEqual(recorder.getTraces().length, 1);
+    t.end();
+});
+
+test('test recording child finishes after parent', (t) => {
+    const recorder = new ListRecorder();
+    const tracer = new APMTracer({ recorder: recorder });
+    const rootSpan = tracer.startSpan('Name');
+    const child = tracer.startSpan('Name', { childOf: rootSpan });
+
+    rootSpan.finish();
+    t.deepEqual(recorder.getTraces().length, 0);
+
+    child.finish();
+    t.deepEqual(recorder.getTraces().length, 1);
+    t.end();
+});
+
+test('test trace serialization, one span', (t) => {
+    const recorder = new ListRecorder();
+    const tracer = new APMTracer({ recorder: recorder });
+    const rootSpan = tracer.startSpan('Name');
+
+    rootSpan.finish();
+    t.deepEqual(recorder.getTraces().length, 1);
+
+    t.deepEquals(recorder.getTraces(), [{
+        traceId: rootSpan.getTraceId(),
+        fragmentId: rootSpan.getId(),
+        timestamp: rootSpan.getStartTime() * 1000,
+        nodes: [{
+            type: 'Component',
+            uri: null,
+            timestamp: rootSpan.getStartTime() * 1000,
+            operation: rootSpan.getOperationName(),
+            duration: rootSpan.getDuration() * 1000,
+        }],
+    }]);
+    t.end();
+});
+
+test('test trace serialization, two spans', (t) => {
+    const recorder = new ListRecorder();
+    const tracer = new APMTracer({ recorder: recorder });
+    const rootSpan = tracer.startSpan('root');
+    const childSpan1 = tracer.startSpan('child1', { childOf: rootSpan });
+    childSpan1.finish(new Date().getTime()  + 150);
+    rootSpan.finish(new Date().getTime()  + 200);
+    t.deepEqual(recorder.getTraces().length, 1);
+
+    t.deepEquals(recorder.getTraces(), [{
+        traceId: rootSpan.getTraceId(),
+        fragmentId: rootSpan.getId(),
+        timestamp: rootSpan.getStartTime() * 1000,
+        nodes: [{
+            type: 'Component',
+            uri: null,
+            timestamp: rootSpan.getStartTime() * 1000,
+            operation: rootSpan.getOperationName(),
+            duration: rootSpan.getDuration() * 1000,
+            nodes: [{
+                type: 'Component',
+                uri: null,
+                timestamp: childSpan1.getStartTime() * 1000,
+                operation: childSpan1.getOperationName(),
+                duration: childSpan1.getDuration() * 1000,
+            },]
+        }],
+    }]);
+    t.end();
+});
+
+test('test trace serialization, three spans', (t) => {
```

Comment:
Would be good to add the TXN ID in this test too - receiving a value in the consumer and then ensuring it is passed on via the producer.